rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -543,6 +546,299 @@ void extractRedeemScriptFromInput_migration_multiple_inputs() {
         }
     }
 
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    @Tag("test sig insertion index and sign input methods for legacy and segwit feds")
+    class Signing {
+        private final List<BtcECKey> federationSignersKeys = BitcoinTestUtils.getBtcEcKeysFromSeeds(new String[]{
+            "member01", "member02", "member03", "member04", "member05", "member06", "member07", "member08", "member09"
+        }, true); // we need the priv keys
+        private final BtcECKey firstFederatorSigningKey = federationSignersKeys.get(0);
+        private final BtcECKey secondFederatorSigningKey = federationSignersKeys.get(1);
+        private final BtcECKey thirdFederatorSigningKey = federationSignersKeys.get(2);
+        private final byte[] emptyByte = new byte[] {};
+        private final int inputIndex = 0; // our tx will just have one input
+
+        private BtcTransaction tx;
+        private List<Sha256Hash> sigHashes;
+        private Sha256Hash sigHash;
+
+        private void setUp(Federation federation) {
+            BtcTransaction prevTx = new BtcTransaction(btcMainnetParams);
+            prevTx.setVersion(ReleaseTransactionBuilder.BTC_TX_VERSION_2);
+            Coin prevValue = Coin.COIN;
+            prevTx.addOutput(prevValue, federation.getAddress());
+
+            tx = new BtcTransaction(btcMainnetParams);
+            tx.setVersion(ReleaseTransactionBuilder.BTC_TX_VERSION_2);
+            tx.addInput(prevTx.getOutput(0));
+
+            addSpendingFederationBaseScript(tx, inputIndex, federation.getRedeemScript(), federation.getFormatVersion());
+
+            sigHashes = generateTransactionInputsSigHashes(tx);
+            sigHash = sigHashes.get(inputIndex);
+        }
+
+        @ParameterizedTest
+        @MethodSource("federationArgs")
+        void getSigInsertionIndex_forFirstAndSecondFederators_shouldReturnCorrectInsertionIndex(Federation federation) {
```

Comment:
To me, this test looks like an IT test. Why not keep it simple with `AAA` structure?